hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -2126,6 +2127,92 @@ private class Emit(
           }
         }
 
+      case x@NDArrayConcat(nds, axis) =>
+        val inputType = coerce[PArray](nds.pType2)
+        val inputNDType = coerce[PNDArray](inputType.elementType)
+
+        val ndType = coerce[PNDArray](x.pType2)
+        val codeNDs = emit(nds, env, er, None)
+
+        val inputArray = mb.newField[Long]
+        val n = mb.newField[Int]
+        val i = mb.newField[Int]
+
+        val loadAndValidateArray = Code(
+          inputArray := codeNDs.value[Long],
+          n := inputType.loadLength(inputArray),
+          (n < 1).orEmpty(Code._fatal("need at least 1 array to concatenate")))
+
+        val (missingSetup: Code[Unit @unchecked], missing: Code[Boolean @unchecked], setupShape: Code[Unit @unchecked]) = (inputType.required, inputNDType.required) match {
```

Comment:
Why do you need these `@unchecked` things? (Not a criticism, I've just not encountered this before so I don't know what it's for)

Reply:
the Scala compiler will complain if you're trying to extract something like this and the internal type is technically unchecked due to type erasure.